esp-hal/src/clock/mod.rs

```diff
@@ -347,7 +347,10 @@ impl Clocks {
                     b"auto" => XtalClock::Other(0), // Can't be `unreachable!` due to const eval.
                     b"26" => XtalClock::_26M,
                     b"40" => XtalClock::_40M,
-                    other => XtalClock::Other(esp_config::esp_config_int_parse!(u32, other)),
+                    _ => XtalClock::Other(esp_config::esp_config_int_parse!(
```

Comment:
Why did you have to change these?

Reply:
`from_str_radix` (and so the macro) takes a `&str` - but we cannot match on `&str` in const

Reply:
Okay but can we have something better, like extracting the config into a variable and reusing that?

Reply:
sure - did that - but in general the code is ugly and probably we could do better here - but not with this PR